pkg/agent/runner.go

```diff
@@ -1476,6 +1500,28 @@ func (s *atomicUpdateState) computeUnitsBounds() (uint16, uint16) {
 	return util.Min(minCPUUnits, minMemUnits), util.Max(minCPUUnits, minMemUnits)
 }
 
+// requiredCUForRequestedUpscaling returns the minimum Compute Units required to abide by the
+// requested upscaling, if there is any.
+//
+// If there's no requested upscaling, then this method will return zero.
+//
+// This method does not respect any bounds on Compute Units placed by the VM's maximum or minimum
+// resource allocation.
+func (s *atomicUpdateState) requiredCUForRequestedUpscaling() uint16 {
+	var required uint16
+
+	// note: floor(x / M) + 1 gives the minimum integer value greater than x / M.
+
+	if s.requestedUpscale.Cpu {
+		required = util.Max(required, s.vm.Cpu.Use/s.computeUnit.VCPU+1)
+	}
```

Comment:
maybe? I definitely had to read it more than times, and cross reference to the definition of `util.Max` which spins over an atomic. I'd feel a bit better about this if the util functions had unittests. 

I think I had to read it more than once because doing a max of two numbers when one is obviously a zero makes me think something fishy is up, or looks like it's unintentional.

Reply:
> `util.Max` which spins over an atomic

I think you're confusing it with `util.AtomicMax`? The definition of `util.Max` is:

https://github.com/neondatabase/autoscaling/blob/bfbf9873c8ea8400536b1b54999c0b421ac48687/pkg/util/arith.go#L19-L26

> doing a max of two numbers when one is obviously a zero makes me think something fishy is up

Fair enough. I think it's personal taste — for me, reading it without a maximum on the first means I have to double-check that the high-level logic on the first & second are still fundamentally the same, even though they're phrased differently.